scripts/top-search-queries/get-top-search-results.go

```diff
@@ -0,0 +1,241 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"io"
+
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ONSdigital/dp-search-api/models"
+	"github.com/ONSdigital/dp-search-api/sdk"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// A searchResponse struct to map the response, from the Search API, to a particular query
+type searchResponse struct {
+	Items []searchItem `json:"items"`
+}
+
+// A SearchItem Struct to map each searchItem of the searchResponse to.
+type searchItem struct {
+	Type        string `json:"type"`
+	Edition     string `json:"edition"`
+	ReleaseDate string `json:"release_date"`
+	Summary     string `json:"summary"`
+	Title       string `json:"title"`
+	Uri         string `json:"uri"`
+}
+
+type runConfig struct {
+	APIURL        string `json:"apiurl"`
+	InputFileName string `json:"inputFileName"`
+	NumResults    string `json:"numResults"`
+}
+
+// main reads in a list of queries from a text file. These are intended to be the queries that are most commonly used
+// in the live Search API service.
+//
+// It then creates an output CSV file and processes the queries, by calling the live Search API, and outputs
+// all the query results to that CSV file.
+//
+// Each query is called with NLP (Natural Language Processing) initially switched off, then called again with
+// NLP switched on. The top 10 results of each query call are added to the output CSV file.
+func main() {
+	ctx := context.Background()
+	log.Info(ctx, "starting script to get results of most common queries from the Search API")
+
+	config := runConfig{}
+	flag.StringVar(&config.APIURL, "api_url", "https://api.beta.ons.gov.uk/v1", "the base url for the search api")
+	flag.StringVar(&config.InputFileName, "input_file_name", "search-queries.txt", "name of input file including extension")
+	flag.StringVar(&config.NumResults, "num_results", "10", "number of results to fetch for each query")
```

Comment:
This should be a number not a string
```suggestion
	flag.IntVar(&config.NumResults, "num_results", "10", "number of results to fetch for each query")
```

Reply:
Thanks Jess, I didn't see the IntVar option previously, that makes sense. Done.